core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/AbstractHealthCheckedEndpointGroupBuilder.java

```diff
@@ -155,12 +162,53 @@ public AbstractHealthCheckedEndpointGroupBuilder withClientOptions(
         return this;
     }
 
+    /**
+     * Sets the maximum endpoint ratio of target selected candidates.
+     * @see PartialHealthCheckStrategyBuilder#maxEndpointRatio(double)
+     */
+    public AbstractHealthCheckedEndpointGroupBuilder maxEndpointRatio(double maxEndpointRatio) {
+        if (partialHealthCheckStrategyBuilder == null) {
+            partialHealthCheckStrategyBuilder = new PartialHealthCheckStrategyBuilder();
+        }
+
+        partialHealthCheckStrategyBuilder.maxEndpointRatio(maxEndpointRatio);
+        return this;
+    }
+
+    /**
+     * Sets the maximum endpoint count of target selected candidates.
+     * @see PartialHealthCheckStrategyBuilder#maxEndpointCount(int)
+     */
+    public AbstractHealthCheckedEndpointGroupBuilder maxEndpointCount(int maxEndpointCount) {
+        if (partialHealthCheckStrategyBuilder == null) {
+            partialHealthCheckStrategyBuilder = new PartialHealthCheckStrategyBuilder();
+        }
+
+        partialHealthCheckStrategyBuilder.maxEndpointCount(maxEndpointCount);
+        return this;
+    }
+
     /**
      * Returns a newly created {@link HealthCheckedEndpointGroup} based on the properties set so far.
      */
     public HealthCheckedEndpointGroup build() {
+        final HealthCheckStrategy healthCheckStrategy =
+                partialHealthCheckStrategyBuilder == null ? this.healthCheckStrategy
```

Comment:
- `this.healthCheckStrategy` -> `HealthCheckStrategy.all()` or just `new All...()`.
- Perhaps we can remove the static factory methods in `HealthCheckStrategy` because they are not used anymore?

Reply:
fixed